### ISSUE TITLE
Analyze SASS files?

### ISSUE THREAD

**Issue by zakjan** [sverweij/dependency-cruiser#53]
With default rules and the latest WebPack support, if a file contains `import "./style.sass"`, this SASS file is also analyzed. But it can contain import paths that are not recognized by WebPack/Depcruise:

`@import "vars"` throws:

```
  error not-to-unresolvable: src/xxx.sass → vars
  error no-non-package-json: src/xxx.sass → vars
```

This can be valid outside of WebPack/Depcruise, because:

- SASS automatically tries sass, scss, css extensions, while these extensions don't need to be in WebPack resolve.extensions config
- sass-loader has separate includePaths option

For now, I excluded SASS files with `"exclude": "\\.s[ac]ss$"`. The question is, should Depcruise improve support for such files, or always exclude them?

Thanks for great project btw!

**Comment by sverweij** [comment]
Hi @zakjan - that's a relevant question.

There's no support for sass/ less/ stylus/ css in dependency-cruiser. Yet. I've tried to focus on getting javascript and typescript to work first, before venturing into other areas. I'm also aware the resolution rules for stylesheet languages differ - which would mean I'd have to find (or write) both a parser and a resolver. Doable, useful (esp in a react context) and fun to build, but a future feature.

In the mean time dependency-cruiser violates the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) when it tries to parse & resolve stuff it doesn't really understand. Expect a PR to hit GitHub in a few days that mends this, so you don't have to set up an `exclude`  or `doNotFollow` rule anymore to prevent false positives. 

The way I propose it will work: if dependency-cruiser encounters a file with an extension it (1) does not know, or (2) knows to be not followable, it will not try to derive new dependencies from it and set the `followable` flag to `false`. (Which is how it works for `json` already).

(And thanks for the kind words :-) )

**Comment by zakjan** [comment]
Yes, with the new version files are ignored correctly.

**Comment by sverweij** [comment]
@zakjan thanks for checking it out. I've merged it and published as latest (`dependency-cruiser@4.3.1`).

Analysing sass (/ less/ stylus/ postcss/ ...) is on the [backlog](https://github.com/sverweij/dependency-cruiser/projects/1#card-11320695).
